### PR TITLE
Document "Inso CLI on docker"

### PR DESCRIPTION
**Pull request by filfreire**
### Summary

This PR documents the how folks can build and run Inso CLI on a Docker container.

### Reason

It's related to this PR: https://github.com/Kong/insomnia/pull/4170

> Note: It's best if we merge it only after we merge the [PR above](https://github.com/Kong/insomnia/pull/4170).

### Testing

I tried out the document on a local instance of insomnia-docs:
Screenshot 2021-11-18 at 18 19 08

cc @develohpanda @falondarville 

Closes INS-1177

### CODE REVIEW

File: docs/inso-cli/inso-on-docker.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+layout: article-detail
+title:  Inso CLI on Docker
+category: "Inso CLI"
+category-url: inso-cli
+---
+
+> Note: this feature is still experimental and in active development.
```
Comment (falondarville) [inline docs/inso-cli/inso-on-docker.md:8]:
```suggestion
{:.alert .alert-primary}
> **Note**: this feature is still experimental and in active development.
```

File: docs/inso-cli/inso-on-docker.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+## Building the `inso` container
```
Comment (falondarville) [inline docs/inso-cli/inso-on-docker.md:10]:
```suggestion
## Build the `inso` Container
```

Using active voice and avoiding "ing".

File: docs/inso-cli/inso-on-docker.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+1. Clone the [insomnia repository](https://github.com/Kong/insomnia).
```
Comment (falondarville) [inline docs/inso-cli/inso-on-docker.md:12]:
```suggestion
1. Clone the [Insomnia repository](https://github.com/Kong/insomnia).
```

File: docs/inso-cli/inso-on-docker.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+2. Install [Docker](https://docs.docker.com/get-docker/).
+3. Run `docker build -t insomnia-inso:dev -f inso-dev.Dockerfile .`.
+
+## Using the `inso` container
```
Comment (falondarville) [inline docs/inso-cli/inso-on-docker.md:16]:
```suggestion
## Use the `inso` Container
```

File: docs/inso-cli/inso-on-docker.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+In order to run Insomnia specs in a container, you'll need to mount the specs folder on your host machine to a given folder on the container. Examples:
```
Comment (falondarville) [inline docs/inso-cli/inso-on-docker.md:18]:
```suggestion
To run Insomnia specs in a Docker container, mount the specs folder on your host machine to a given folder on the container. Examples:
```

Increase casual language and specificity.

File: docs/inso-cli/inso-on-docker.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+- Mounting an Insomnia git sync repository folder to a folder on the container:
```
Comment (falondarville) [inline docs/inso-cli/inso-on-docker.md:20]:
```suggestion
- Mount an Insomnia git sync repository folder to a folder on the container:
```

File: docs/inso-cli/inso-on-docker.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+```shell
+cd <your-git-sync-repo-folder>
+
+docker run -it --rm -v $(pwd):/var/temp insomnia-inso:dev run test -w /var/temp
+```
+
+- Mounting the Insomnia Application Data folder:
```
Comment (falondarville) [inline docs/inso-cli/inso-on-docker.md:28]:
```suggestion
- Mount the Insomnia Application Data folder:
```

File: docs/inso-cli/inso-on-docker.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+```shell
+# On macOS:
+
```
Comment (falondarville) [inline docs/inso-cli/inso-on-docker.md:32]:
```suggestion
```

File: docs/inso-cli/inso-on-docker.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+docker run -v $HOME/Library/Application\ Support/Insomnia:/var/temp -it --rm insomnia-inso:dev run test --src /var/temp
+
+# On Linux:
+docker run -v $HOME/.config/Insomnia:/var/temp -it --rm insomnia-inso:dev run test --src /var/temp
+
+# On Windows (using Docker for Windows and WSL):
+docker run -v /mnt/c/Users/<your_username>/AppData/Roaming/Insomnia:/var/temp -it --rm insomnia-inso:dev run test --src /var/temp
+```
+
+- Mounting the folder where you keep an Insomnia v4 export:
```
Comment (falondarville) [inline docs/inso-cli/inso-on-docker.md:42]:
```suggestion
- Mount the folder where you keep an Insomnia v4 export:
```

**Review by falondarville — APPROVED** [review]
Thank you so much, Filipe. Would you accept my changes? They are just topical style guide updates.

**Comment by develohpanda** [comment]
Hey @filfreire and @falondarville, this PR is specifically for using docker for inso development, rather than consuming the published versions.

These are DX docs, but we do need something very similar for production docs, based on [this](https://kongstrong.slack.com/archives/C02F6MVBCA3/p1637248640036600?thread_ts=1636729070.013300&channel=C02F6MVBCA3&message_ts=1637248640.036600) Slack thread.

I think the only thing that needs to change for converting these DX docs to consuming the production binaries is the top section of the page, and the DX docs themselves can remain within the insomnia repository. Thoughts?

**Comment by filfreire** [comment]
@develohpanda we could do this in multiple ways:

- we share this "now" and afterwards, once we start pushing production docker images of inso - edit the doc so folks don't build this locally - they could simply do `docker pull insomnia-inso:prod` (or `insomnia-inso:latest`, `insomnia-inso:2.4.0`, ...), and then run the same `docker run ...` commands.

or,

- we wait until we are officially pushing production docker images, edit this by then, and make it share it afterwards

As for DX stuff you mentioned, I removed it from my other PR because I noticed we were making `inso` README more lean, and making the docs here the source of truth. I'm ok with re-adding the info there, or keeping the docs here as the source of truth. Would like to get @falondarville take on this.

**Comment by falondarville** [comment]
@develohpanda do you think this content should live in the Insomnia README instead?

**Comment by develohpanda** [comment]
> I removed it from my other PR because I noticed we were making inso README more lean, and making the docs here the source of truth

The Inso README was split in two: the usage/production docs were moved to the docs site in https://github.com/Kong/insomnia/pull/4169, and the _development_ docs were moved to the README at the root of the repository in https://github.com/Kong/insomnia/pull/4224, to live alongside the insomnia-app development docs.

> I think we can keep it on the docs site. We'll rename the file to show that this is for development only

Because we don't have any other dev docs on the public docs site I don't think we should have the Inso dev docs here either. It seems standard practice to keep development docs alongside the code within the repository, and user docs on a docs site. Thoughts?

---

This page is still needed as per INS-1177, but just needs to be adjusted for scope to use the inso single executable as an example dockerfile and show users how to use an Inso docker image in production. We _could_ skip that entirely if we wait until official docker production images are being shipped, too; saves a lot of work, but that's not planned work right now AFAIK.

**Comment by falondarville** [comment]
> Because we don't have any other dev docs on the public docs site I don't think we should have the Inso dev docs here either. It seems standard practice to keep development docs alongside the code within the repository, and user docs on a docs site. Thoughts?

We do have this [dev doc](https://docs.insomnia.rest/insomnia/custom-themes) for custom themes on the docs site. Not that this should be the measure of where future dev docs should live. Let's discuss as this is generally a larger docs question that will determine our future course/params.

**Comment by develohpanda** [comment]
> We do have this dev doc for custom themes on the docs site.

Resolved on a call but noting here for completeness; those are plugin docs intended for use by plugin developers, so technically it is developer documentation, but differs from dev docs that exist in the repository readme

**Comment by filfreire** [comment]
@develohpanda @falondarville follow-up to our discussions, going to close this one for now, we'll reopen and edit once we officially publish inso-cli docker images.

**Comment by develohpanda** [comment]
Sounds good - added INS-1238 to track.

**Pull request closed without merging**